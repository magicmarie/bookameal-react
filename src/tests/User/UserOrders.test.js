import React from "react";
import { shallow } from "enzyme";
import MockAdapter from "axios-mock-adapter";
import axiosInstance from "../../components/common/Apicalls";
import UserOrders from "../../components/User/UsersOrders";

describe("AdminMenuCard component", () => {
  let data;
  let wrapper;
  beforeEach(() => {
    data = {
      currentPage: 1,
      orders: [{ id: 1, meal_name: "test", price: 1000 }],
      nextPage: 2,
      pages: 2,
      previousPage: ""
    };
    wrapper = shallow(<UserOrders />);
  });
  it("renders successfully", () => {
    const axiosMock = new MockAdapter(axiosInstance);
    axiosMock.onGet("/user/orders").reply(200, {
      Orders: data
    });
  });

  it("fails successfully", () => {
    const axiosMock = new MockAdapter(axiosInstance);
    axiosMock.onGet("/user/orders").reply(401, {
      Orders: data
    });
  });
  it("changes page", () => {
    const { currentPage } = wrapper.instance().state;
    expect(currentPage).toBe(null);
    wrapper.instance().changePage(1);
    expect(wrapper.instance().state.currentPage).toBe(1);
  });
  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
