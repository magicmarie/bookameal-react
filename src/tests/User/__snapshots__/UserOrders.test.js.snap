// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminMenuCard component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserOrders />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          component="div"
          loaded={false}
          loadedClassName="loadedContent"
          parentClassName="loader"
>
          <h2
                    className="header text-center"
          >
                    Orders
          </h2>
          <div
                    className="row"
          >
                    <div>
                              No orders Found
                    </div>
          </div>
</Unknown>,
        <Pagination
          changePage={[Function]}
          currentPage={null}
          nextPage={null}
          pages={null}
          previousPage={null}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <h2
              className="header text-center"
>
              Orders
</h2>,
            <div
              className="row"
>
              <div>
                            No orders Found
              </div>
</div>,
          ],
          "component": "div",
          "loaded": false,
          "loadedClassName": "loadedContent",
          "parentClassName": "loader",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Orders",
              "className": "header text-center",
            },
            "ref": null,
            "rendered": "Orders",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div>
                No orders Found
</div>,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No orders Found",
              },
              "ref": null,
              "rendered": "No orders Found",
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "changePage": [Function],
          "currentPage": null,
          "nextPage": null,
          "pages": null,
          "previousPage": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            component="div"
            loaded={false}
            loadedClassName="loadedContent"
            parentClassName="loader"
>
            <h2
                        className="header text-center"
            >
                        Orders
            </h2>
            <div
                        className="row"
            >
                        <div>
                                    No orders Found
                        </div>
            </div>
</Unknown>,
          <Pagination
            changePage={[Function]}
            currentPage={null}
            nextPage={null}
            pages={null}
            previousPage={null}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <h2
                className="header text-center"
>
                Orders
</h2>,
              <div
                className="row"
>
                <div>
                                No orders Found
                </div>
</div>,
            ],
            "component": "div",
            "loaded": false,
            "loadedClassName": "loadedContent",
            "parentClassName": "loader",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Orders",
                "className": "header text-center",
              },
              "ref": null,
              "rendered": "Orders",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div>
                  No orders Found
</div>,
                "className": "row",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "No orders Found",
                },
                "ref": null,
                "rendered": "No orders Found",
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "changePage": [Function],
            "currentPage": null,
            "nextPage": null,
            "pages": null,
            "previousPage": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
